lib/src/components/stack/Ancestor_Stack.res

```diff
@@ -1 +1,715 @@
+module Make = (Maker: Ancestor_StylesMaker.T) => {
+  module Styles = Ancestor_Styles.Make(Maker)
+  module Base = Ancestor_Base.Make(Maker)
 
+  type direction = [
+    | #vertical
+    | #horizontal
+  ]
+
+  /**
+   * Specific styles for the API.
+   */
+  let createStack = (~direction=?, ~spacing=?, ()) => {
+    let baseStyles = `
+      display: flex;
+    `
+
+    let responsiveStyles = [
+      Styles.createResponsiveProp(~prop=spacing, spacing =>
+        `gap: ${Styles.Spacing.make(spacing)};`
+      ),
+      Styles.createResponsiveProp(
+        ~prop=direction,
+        ~defaultStyles=`flex-direction: row;`,
+        direction =>
+          switch direction {
+          | #vertical => "flex-direction: row;"
+          | #horizontal => "flex-direction: column"
+          },
+      ),
+    ]
+
+    Maker.css(
+      `
+      ${baseStyles}
+      ${responsiveStyles->Styles.merge}
+      `,
+    )
+  }
+
+  let renderWithDivider = (children, divider) => {
+    let total = React.Children.count(children)
+    children->React.Children.mapWithIndex((element, index) => {
+      if index == total - 1 {
+        element
+      } else {
+        <> element divider </>
+      }
+    })
+  }
+
+  @react.component
+  let make = (
+    // Stack props
+    ~direction: option<Styles.responsiveProp<direction>>=?,
+    ~spacing: option<Styles.responsiveProp<Styles.Spacing.t>>=?,
+    ~divider: option<React.element>=?,
```

Comment:
IMO dividers should take 0 height. I wonder if the library should ensure that is the case? Or if this responsibility to deferred to the consumer?

I suppose if you're using ancestor to create your design system then you'd end up wrapping Stack and making this a bool anyway, so giving some control here is a good thing.

Reply:
> IMO dividers should take 0 height

I'm not sure about that. Can you elaborate more?

> I suppose if you're using ancestor to create your design system then you'd end up wrapping Stack and making this a bool anyway, so giving some control here is a good thing.

Yep, when I was creating a draft I thought in some cases where you can use Stack as a component to group elements in a screen, sometimes you have a divider already created on your project (design system) but sometimes the design team create an specific component that solves a specific problem in the UI, so I think is a bit of flexibility, in this case, is good.

Reply:
If dividers are non-zero height then adding them to your stack will break your grid. Imagine if I want "4 grid units" between each element, and my divider has the height of 0.5 grid units - my stack now has 4.5 units of space instead of 4. This breaks the contract of the Stack `spacing` prop IMO.

Reply:
I'm not sure If we're on the same page in this case. In this current implementation, at least, the stack doesn't work like a grid. It just stack the elements, if you add a divider with a height of `16px` the container height/width will be increased and everything works (of course, you have to worry about the scroll and wrap behavior).
Am I misunderstanding?